Bind monster stat points between 20 and 100, fix target handling in Ring of Flames, fix client crash with colored orc shield. Misc fixes.

skittles1 opened the pull request:
Bugfix PR. Everything here is live on 105.
##### Commit 1

Monster stat point allocation must use values lower than 100 otherwise
the math does not work and stats are incorrectly set. This can occur for
difficulty 10 mobs.
##### Commit 2

Ring of Flames now correctly handles casting on self, and cast messages
are correctly displayed when casting on others.
##### Commit 3

Orc shield color translation was being sent after animation/group data
instead of before. Shield handles this correctly.
##### Commit 4

Fix CountMonsters call in RatInvasion - should be called on the event room, with mob class.
##### Commit 5

Treasure generation should not call SpellItemIsAccessible on every replacement item for an inaccessible SpellItem, so check class of replacement item first.
##### Commit 6

Fix mismatched player default nose/mouth resources.Affected players which were not yet created (<New Character> slots) which could be used for statues in Hall of Forgotten Heroes. Run "send c player FixDefaultResources" after updating with this commit to fix the issue.
##### Commit 7

Fix eavesdrop yell list handling. Yell list checks were using room number and room object interchangeably. Caused errors when used in some rooms.

skittles1 committed:
Bind monster stat points between 20 and 100.

Monster stat point allocation must use values lower than 100 otherwise
the math does not work and stats are incorrectly set. This can occur for
difficulty 10 mobs.

skittles1 committed:
Fix target handling in Ring of Flames.

Now correctly handles casting on self, and cast messages are correctly
displayed when casting on others.

skittles1 committed:
Fix client crashing bug with colored orc shields.

Color translation was being sent after animation/group data instead of
before. Shield handles this correctly.

treymd commented:
Should probably avoid stacking unrelated commits in a single pull request unless necessary.  Even if these are all perfect it is a bad precedent to set.

skittles1 commented:
These are bugfixes - the alternative is I don't submit them.

skittles1 committed:
Fix CountMonsters call in RatInvasion event.

Should be called on room, with mob class.

skittles1 committed:
Only check SpellItems for accessibility when generating treasure.

All replacement items were being checked, which logs an error when the
message isn't handled.

skittles1 committed:
Fix player default nose/mouth resources.

Affected players which were not yet created (<New Character> slots)
which could be used for statues in Hall of Forgotten Heroes. Run "send c
player FixDefaultResources" after updating with this commit to fix the
issue.

skittles1 committed:
Fix eavesdrop yell list handling.

Yell list checks were using room number and room object interchangeably.
Caused errors when used in some rooms.
